firmware/src/dap.rs

```diff
@@ -472,40 +478,58 @@ impl<'a> DAP<'a> {
         let mask = req.next_u8();
         let wait = req.next_u32();
 
-        // Our pin mapping:
-        // SWDIO/TMS: FLASH_SI
-        // SWCLK/TCK: SCK
-        // SWO/TDO: FLASH_CS
-        // TDI: FPGA_RST
-        // nRESET: FLASH_SO
-        //
         // SWJ_Pins mapping:
         // 0: SWCLK/TCK
         // 1: SWDIO/TMS
         // 2: TDI
         // 3: TDO
         // 5: nTRST
         // 7: nRESET
-        //
-        // We only support setting nRESET.
+
+        self.pins.reset.set_mode_output();
+        self.pins.usart1_rx.set_mode_output();
+        self.pins.spi2_clk.set_mode_output();
+        self.pins.spi2_miso.set_mode_output();
```

Comment:
This looks so wrong... I wonder how long it takes to destroy your chip or its pins.

Reply:
Well, one can set output direction only if requested, not every time. And probably for MISO and other input lines we can ignore direction since setting them to output is dangerous anyway.

Reply:
Makes sense. What would be a sensible way of restoring GPIO modes expected by DAP commands that follow? Otherwise a state of a DAP session state machine is violated and commands' assumptions about pin modes are not upheld. Matching on mode in `DAP::process_command` and enforcing pin mode is definitely bad.

Reply:
It's not clear how it should work in practice and when probe should restore its pin states. That's why only nRESET was implemented: it's a regular GPIO, not something driven by SPI/UART. Even switching between two SPIs doesn't work reliably on the chip we use, so it's better to avoid switching. What is your use case by the way?

Reply:
I updated the PR description. This is definitely not finished, I mostly aim to have something to continue the discussion of how this can be pushed forward.